### PR TITLE
Like enhancements unit testing 2

### PR DESCRIPTION
**Pull request by develric**
This is a second part PR to the #14532. Adding unit testing to view model layers for likes enhancements.

## To test
To test it should be enough to check the CI is green.

## Regression Notes
N/A; behind feature flag.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/engagement/EngagedPeopleListViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+import org.wordpress.android.ui.pages.SnackbarMessageHolder
+import org.wordpress.android.ui.reader.utils.ReaderUtilsWrapper
+import org.wordpress.android.util.analytics.AnalyticsUtilsWrapper
+import org.wordpress.android.viewmodel.Event
+
+@InternalCoroutinesApi
+@RunWith(MockitoJUnitRunner::class)
+class EngagedPeopleListViewModelTest {
+    @Rule
+    @JvmField val rule = InstantTaskExecutorRule()
```
Comment (khaykov) [inline WordPress/src/test/java/org/wordpress/android/ui/engagement/EngagedPeopleListViewModelTest.kt]: FYI, we have a `BaseUnitTest` class that you can extend, and it has both `@RunWith(MockitoJUnitRunner::class)` and `@Rule @JvmField val rule = InstantTaskExecutorRule()` :)

File: WordPress/src/test/java/org/wordpress/android/ui/engagement/EngagedPeopleListViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+
+        viewModel.start(listScenario)
+
+        getLikesState.value = likesState
+
+        requireNotNull(uiState).let {
+            assertThat(it.showLikeFacesTrain).isFalse
+            assertThat(it.showLoading).isFalse
+            with(it.engageItemsList) {
+                val likedItem = this.filterIsInstance<LikedItem>()
```
Comment (khaykov) [inline WordPress/src/test/java/org/wordpress/android/ui/engagement/EngagedPeopleListViewModelTest.kt:200]: TIL about `filterIsInstance` !

File: WordPress/src/test/java/org/wordpress/android/ui/engagement/utils/GetLikesTestUtils.kt [hunk 3]
```diff
@@ -30,6 +49,61 @@ fun getDefaultLikers(numLikers: Int, likedItemType: LikeType, siteId: Long, item
...
+    TEST_CONFIG_1,
+    TEST_CONFIG_2,
+    TEST_CONFIG_3,
+    TEST_CONFIG_4,
+    TEST_CONFIG_5
+}
+
+fun getGetLikesState(testConfig: GetLikesTestConfig): GetLikesState {
+    return when (testConfig) {
+        TEST_CONFIG_1 -> {
```
Comment (khaykov) [inline WordPress/src/test/java/org/wordpress/android/ui/engagement/utils/GetLikesTestUtils.kt:63]: What do you think about adding a comment to each state explaining briefly what it represents?

File: WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt [hunk 4]
```diff
@@ -772,6 +791,80 @@ class ReaderPostDetailViewModelTest {
...
+        val post = mock<ReaderPost>()
+        whenever(post.isWP).thenReturn(true)
+        viewModel.post = post
+        val likeObserver = init().likesUiState
+
+        getLikesState.value = likesState
+
+        assertThat(likeObserver).isNotEmpty
+        with(likeObserver.first()) {
+            assertThat(showLikeFacesTrain).isTrue
```
Comment (khaykov) [inline WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt:839]: Not sure, but shouldn't `showLikeFacesTrain` be false on failure and no cache?

**Review by khaykov — COMMENTED** [review]
Thanks for all the tests, @develric ! Looks good to me, I just left a couple of non-critical questions/comments :)

File: WordPress/src/test/java/org/wordpress/android/ui/engagement/EngagedPeopleListViewModelTest.kt (hunk 1 above)
Comment (develric) [reply WordPress/src/test/java/org/wordpress/android/ui/engagement/EngagedPeopleListViewModelTest.kt]: ahah 😄 ! Now that you say it, I think I even knew about it before but totally forgot! Changed to use it 🙇 .

File: WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt (hunk 4 above)
Comment (develric) [reply WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt:839]: So, I think we should rename the `showLikeFacesTrain` to something like `showLikeFacesTrainContainer`; it is actually the root container so it should be visible to show the empty state, but agree the naming is not good 👍 . I'm going to open a small separate PR for that if you agree 🙇

Comment (khaykov) [reply WordPress/src/test/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModelTest.kt:839]: Sounds good 👍

**Review by khaykov — APPROVED** [review]
Thanks for the changes! Looks good to me 🚀

**Pull request merged**